### PR TITLE
Protobuf changes for Approve/Allowance

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
- Add new basic types for granted allowances used in AccountInfo
- Remove `PAYER_AND_OWNER_NOT_EQUAL` , since the implementation will support pay to not be same as owner

### CODE REVIEW

**Review by superboo — APPROVED** [review]
LGTM

File: services/basic_types.proto [hunk 1]
```diff
@@ -1592,3 +1592,65 @@ message TokenAllowance {
...
+     * The list of serial numbers that the spender is permitted to transfer.
+     */
+    repeated int64 serialNumbers = 3;
+
+    /**
+     * If true, the spender has access to all of the account owner's NFT instances (currently
+     * owned and any in the future). If this field is set to true the serialNumbers field
+     * should be empty.
+     */
+    google.protobuf.BoolValue approvedForAll = 4;
```
[the remark this answers is not in the document]
Comment (Neeharika-Sompalli) [reply services/basic_types.proto]: @steven-sheehy  fixed to use `bool`

**Review by steven-sheehy — APPROVED** [review]

File: services/basic_types.proto [hunk 2]
```diff
@@ -1592,3 +1592,65 @@ message TokenAllowance {
...
+}
+
+/**
+ * A granted allowance of non-fungible token transfers for a spender relative to the owner account.
+ */
+message GrantedNftAllowance {
+    /**
+     * The token that the allowance pertains to.
+     */
+    TokenID tokenId = 1;
```
Comment (tinker-michaelj) [inline services/basic_types.proto:1618]: Should we use `token_id` here? (And for `serial_numbers`, `approved_for_all`, etc below.)

**Review by SimiHunjan — DISMISSED** [review]

**Review by SimiHunjan — DISMISSED** [review]

File: services/basic_types.proto (hunk 2 above)
Comment (Neeharika-Sompalli) [reply services/basic_types.proto]: @tinker-michaelj  Converted to snake case

**Review by steven-sheehy — APPROVED** [review]

**Review by SimiHunjan — APPROVED** [review]

**Pull request merged**